tests/test_pixel_models.py

```diff
@@ -0,0 +1,102 @@
+from nexus_constructor.qml_models.pixel_models import (
+    PixelGridModel,
+    PixelMappingModel,
+    PixelMapping,
+    SinglePixelModel,
+    SinglePixelId,
+    InstrumentModel,
+    PixelGrid,
+)
+
+
+def test_GIVEN_nothing_WHEN_creating_single_pixel_model_THEN_pixel_id_is_zero():
+    model = SinglePixelModel()
+    assert model.get_pixel_id() == 0
+
+
+def test_GIVEN_pixel_id_WHEN_creating_single_pixel_model_THEN_pixel_id_is_set_in_model():
+    custom_id = 5
+    model = SinglePixelModel()
+    model.set_pixel_id(custom_id)
+    assert model.get_pixel_id() == custom_id
+
+
+def test_GIVEN_pixel_model_WHEN_creating_pixelgridmodel_THEN_model_is_updated():
+    pixel_data = PixelGrid(rows=5)
+    model = PixelGridModel()
+    inst = InstrumentModel()
+    inst.components[0].pixel_data = pixel_data
```

Comment:
Don't assume inst.components[0] exists or that you are allowed to set pixel_data on it. Add a new detector component and set the pixel_data on that.